src/Entity/Produto.php

```diff
@@ -162,7 +150,7 @@ public function jsonSerialize(): array
             'valorUnitario' => $this->getValorUnitario(),
             'criadoEm' => $this->getCriadoEm(),
             'atualizadoEm' => $this->getAtualizadoEm(),
-            'categoria' => $this->getCategoriaId()->getNome(),
+            'categoria' => $this->getCategoria()?->getNome(),
```

Comment:
produto sempre tem um categoria
```suggestion
            'categoria' => $this->getCategoria()->getNome(),
```